src/server/src/main.rs

```diff
@@ -1,23 +1,215 @@
+use bastion::prelude::*;
 use ntex::web;
+use ntex::web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
 
-#[web::get("/")]
-async fn index() -> impl web::Responder {
-    "Hello, World!"
+// user
+#[derive(Deserialize, Serialize, Clone)]
+struct User 
+{
+    username: String,
+    password: String,
+    email: Option<String>,
+    age: Option<u32>,
+    gender: Option<String>,
 }
+// sessionId -> child process reference
+type SessionMap = Arc<Mutex<HashMap<String, ChildRef>>>;
+// username -> User
+type UserMap = Arc<Mutex<HashMap<String, User>>>;
```

Comment:
as mentioned further down, having a thin abstraction over these would be nice. some constructors and getters would be good. once we've abstracted out, we can start doing things like remapping dead session id's as we detect them.